packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -176,60 +179,101 @@ Follows the [Column Type Philosophy](/packages/nimble-components/src/table/specs
+-   `alignment` - a string value matching `"left"`, `"right"`, or `undefined` (the default, meaning `"automatic"`) which controls whether values and column headers are left or right aligned within the column. If set to `undefined` Nimble will choose left or right based on the value of `format`. Clients should select `right` if it is known that the decimal separators of all values in the column will align in the given the `format`.
+-   `format` - a string which controls how the number is formatted for display. It can take one of the following values:
+    -   `undefined` - use a default formatter, which will format similarly to [`Number.toString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString#description). This displays integers with no trailing zeros, limits to about 16 significant digits, and switches to exponential notation for very large and small numbers. Instead of using `toString()` it will be implemented using `Intl.NumberFormat` to achieve more consistent i18n (`toString` always uses a `.` separator and displays the English word for "Infinity"). Will be displayed left-aligned by default (since numbers will display an inconsistent number of fractional digits).
+    -   `'integer'` - format all values as integers, rounding to nearest if the value isn't an integer and never displaying exponential notation. Will be displayed right-aligned by default.
+    -   `'decimal'` - format all values as decimal values (e.g. 123.45), always displaying `decimal-digits` digits after the separator and never displaying exponential notation. Will be displayed right-aligned by default.
+    -   `'custom'` - use [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat) configured with values specified in other attributes. Will be displayed left-aligned by default (since Nimble can't know if the provided configuration will produce consistent decimal separators).
+    -   This could be extended to other pre-configured formats in future. Their configuration attributes would similarly be prefixed with the name of the format mode.
+    -   **Note:** all of the above will be implemented using a `Intl.NumberFormat` formatter. For all modes besides `custom` Nimble will configure the formatter with defaults to match the [visual design spec](https://github.com/ni/nimble/issues/887) (e.g. `useGrouping: false` to achieve `1000` rather than `1,000` and `signDisplay: auto` to achieve `1` rather than `+1`).
+-   `locales` - a string containing a comma-separated list of locales to pass to the [`locales` parameter of the `Intl.NumberFormat` constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locale_identification_and_negotiation) which is used by all format modes. This means numbers will be formatted with locale-specific decimal separators.
```

Comment:
Here's my stab at a proposal (haven't updated spec yet since it seems worth discussing first):
1. within the column element, use `this.closest('[lang]')?.lang` to pick the locale to pass when constructing the formatter (likely in `onConnected()` or whenever a relevant attribute changes)
   - this walks up the DOM tree to find the closest parent (or the column element itself) with a lang attribute. This [seems to be the intended meaning](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang) of the attribute, which is available on every element. It allows any element to override the page's lang.
   - if nothing specifies lang, this will result in `undefined` which the formatter interprets as "use the system language".
2. Don't bother subscribing to [`window.languagechange`](https://developer.mozilla.org/en-US/docs/Web/API/Window/languagechange_event). This is mostly a choice to reduce scope; we could add this in future if needed.
3. Provide clear documentation of the above and emphasize that clients should set a locale (preferably including both language and region) on their `<html>` element.

Some possible gotchas:
 - is it ok to start out forcing clients to refresh the page or poke the column element to detect a language change?
 - is this good enough for Blazor?
 - is this OK for Angular, where [LOCALE_ID](https://angular.io/api/core/LOCALE_ID) is more commonly used to configure localization of pipes?

Reply:
> within the column element, use this.closest('[lang]')?.lang to pick the locale to pass when constructing the formatter (likely in onConnected() or whenever a relevant attribute changes)

Ahhh so with this:

> Maybe this should behave more like the dir attribute? Check the root html element if lang is set to use as the dynamic default value (or just en-us until the dyntamic theme provider default spec is in) and you set lang on theme-provider to override it for parts of the tree (instead of locales).

I was poorly expressing that I think we should handle locale like we are handling direction. i.e. it is a configuration design token that is configured via the theme-provider. The theme-provider would be responsible for checking for the lang of the page. The component would only be using the design token.

The benefit of the token is the reactivity. We can't really easily respond to attributes being set on arbitrary parents we don't own. But we can observe changes to the token. The trade-off is that we have a semi-proprietary api for configuring lang and dir for regions of the page. Nimble users must use the theme-provider to configure those properties different for regions of the page.

```html
<nimble-theme-provider locale="en-us" direction="ltr">
    <nimble-table></nimble-table>
</nimble-theme-provider>
```

Although as I write this I'm wondering if we should use the html attributes so that way the DOM in those regions is configured in the same mode, ie:

```html
<nimble-theme-provider lang="en-us" dir="ltr">
    <nimble-table></nimble-table>
</nimble-theme-provider>
```

Reply:
After an offline conversation with Milan we decided to call the design token `lang` to match the HTML attribute. I update the spec to capture this.